src/main/java/com/github/martynagil/drugstoremanagement/service/ComplaintService.java

```diff
@@ -27,13 +27,13 @@ public ComplaintService(ComplaintRepository complaintRepository, ProductReposito
         this.transactionRepository = transactionRepository;
     }
 
+    @Transactional
     public void addComplaint(ComplaintDto complaintDto) {
         if (complaintExists(complaintDto)) {
             throw new ComplaintAlreadyExistsException();
         }
 
         Complaint complaint = createComplainFromDto(complaintDto);
-        complaintRepository.save(complaint);
```

Comment:
czemu to usunęłaś? Mówiłem że transakcje działają na modyfikacje instniejących encji a nie dodawanie nowych. Framework nie zgadnie za Ciebie czy chcesz zapisać encję czy nie